revision.c

```diff
@@ -109,6 +109,9 @@ void mark_trees_uninteresting_sparse(struct repository *r,
 	while ((oid = oidset_iter_next(&iter))) {
 		struct tree *tree = lookup_tree(r, oid);
 
+		if (!tree)
```

Comment:
maybe i missed something, but i didn't see where anything other than a tree gets added to this set.

Reply:
unfortunately, dropping this can cause failures in some strange cases (like shallow clones). Running the tests with GIT_TEST_PACK_SPARSE=1 demonstrate the problem.